client/me/developer/main.jsx

```diff
@@ -52,43 +63,56 @@ class Developer extends Component {
 	};
 
 	showDevSurveyNotice = () => {
-		const noticeMessage = this.props.translate(
-			"Hey developer! How do {{i}}you{{/i}} use WordPress.com? Spare a moment? We'd love to hear what you think in a {{surveyLink}}quick survey{{/surveyLink}}.",
-			{
-				components: {
-					i: <i />,
-					surveyLink: (
-						<a
-							href={ this.getSurveyHref() }
-							target="_blank"
-							rel="noopener noreferrer"
-							onClick={ () => {
-								recordTracksEvent( 'calypso_me_developer_survey_clicked' );
-
-								this.props.removeNotice( this.developerSurveyNoticeId );
-							} }
-						/>
-					),
-				},
+		if ( this.props.currentUser.localeSlug === 'en' ) {
+			if ( ! this.surveyNoticeWrapper ) {
+				this.surveyNoticeWrapper = document.createElement( 'div' );
+				document.body.appendChild( this.surveyNoticeWrapper );
 			}
-		);
 
-		const noticeProps = {
-			id: this.developerSurveyNoticeId,
-			isPersistent: true,
-			onDismissClick: () => {
-				recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
-
-				this.setDeveloperSurveyCookie( 'dismissed', 24 * 60 * 60 ); // 1 day
-
-				this.props.removeNotice( this.developerSurveyNoticeId );
-			},
-		};
+			this.setState( { isDeveloperSurveyNoticeVisible: true } );
+		} else {
+			const noticeMessage = this.props.translate(
+				"Hey developer! How do {{i}}you{{/i}} use WordPress.com? Spare a moment? We'd love to hear what you think in a {{surveyLink}}quick survey{{/surveyLink}}.",
+				{
+					components: {
+						i: <i />,
+						surveyLink: (
+							<a
+								href={ this.getSurveyHref() }
+								target="_blank"
+								rel="noopener noreferrer"
+								onClick={ () => {
+									recordTracksEvent( 'calypso_me_developer_survey_clicked' );
+
+									this.hideDevSurveyNotice();
+								} }
+							/>
+						),
+					},
+				}
+			);
+
+			const noticeProps = {
+				id: this.developerSurveyNoticeId,
+				isPersistent: true,
+				onDismissClick: () => {
+					recordTracksEvent( 'calypso_me_developer_survey_dismissed' );
+
+					this.setDeveloperSurveyCookie( 'dismissed', 24 * 60 * 60 ); // 1 day
+					this.props.removeNotice( this.developerSurveyNoticeId );
+				},
+			};
+			this.props.successNotice( noticeMessage, noticeProps );
+		}
 
-		this.props.successNotice( noticeMessage, noticeProps );
 		recordTracksEvent( 'calypso_me_developer_survey_impression' );
 	};
 
+	hideDevSurveyNotice = () => {
+		this.setState( { isDeveloperSurveyNoticeVisible: false } );
+		this.props.removeNotice( this.developerSurveyNoticeId );
+	};
+
```

Comment:
So we follow the same approach everywhere:
```suggestion
	hideDeveloperSurveyNotice = () => {
		this.setState( { isDeveloperSurveyNoticeVisible: false } );
		this.props.removeNotice( this.developerSurveyNoticeId );
	};

```

Reply:
Actually there are other places we we could do the same thing (e.g. `shouldShowDevSurveyNotice`).